grype/version/gemfile_version.go

```diff
@@ -0,0 +1,59 @@
+package version
+
+import (
+	"fmt"
+	"strings"
+)
+
+type gemfileVersion struct {
+	// keeping the raw version for transparency,
+	// but its value might change to fit semVer standards: https://semver.org/
+	raw    string
+	semVer *semanticVersion
+}
+
+// Gemfile.lock versions may have "{cpu}-{os}" or "{cpu}-{os}-{version}"
+// after the semvVer, for example, 12.2.1-alpha-x86_64-darwin-8, where `2.2.1-alpha`
+// is a valid and comparable semVer, and `x86_64-darwin-8` is not a semVer due to
+// the underscore. Also, we can't sort based on arch and OS in a way that make sense
+// for versions. SemVer is a characteristic of the code, not which arch OS it runs on.
+//
+// CPU is the most structured value present in gemfile.lock versions, we use it
+// to split the version info in half, the first half has semVer, and
+// the second half has arch and OS which we ignore.
+func extractSemVer(raw string) string {
+	lower := strings.ToLower(raw)
```

Comment:
do we need to convert the entire version to lowercase? will this make a difference in comparison?

Reply:
The arch case is consistent, so no need to lower it. The case would make a different version object cause and comparison. I removed the lowercase step.

Reply:
In constraint comparison that wouldn't make difference.

Reply:
I didn't review the `anchore/go-version` lib to see what the sensitive factors to comparisons are, but I think there is a risk that `1.0.3-alpha` and `1.0.3-ALPHA` will not be considered equal --I think it's good that the `ToLower` has been removed in the current state of the PR for this reason. 

However, if there is a concern of casing, the existing `extractSemVer` could be refactored to split in a case-insensitive way, but always return the substring with no casing changes.

To be clear, I'm not asking for this change (no need to add this to the PR).